File: steam_app/lib/model/objects/Game.dart
class Game {
  int id;
  String name;
  String genre;
  double price;
  int quantity;



  Game({this.id, this.name, this.genre, this.quantity, this.price});

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      id: json['id'],
      name: json['name'],
      genre: json['genre'],
      price: json['price'],
      quantity: json['quantity'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'genre': genre,
    'price': price,
    'quantity':quantity,
  };

  @override
  String toString() {
    return name;
  }


}

File: Backend_PSW_Spring/src/main/java/com/example/steamapp/entities/Game.java
package com.example.steamapp.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@ToString
@EqualsAndHashCode
@Entity
@Table(name = "game", schema = "my_database")

public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private int id;


    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "genre")
    private String genre;

    @Basic
    @Column(name = "price")
    private float price;

    @Basic
    @Column(name = "quantity")
    private int quantity;

    @Version
    @JsonIgnore
    @ToString.Exclude
    @Column(name = "version")
    private long version;


    @OneToMany(mappedBy = "game")
    @JsonIgnore
    @ToString.Exclude
    private List<GamePerOrder> gamePerOrder;

}
